code/modules/http/http.dm

```diff
@@ -4,85 +4,160 @@
+	/// If present, response body will be saved to this file.
 	var/output_file
-
+	/// The raw response, which will be decoeded into a [/datum/http_response]
 	var/_raw_response
-
+	/// Callback for executing after async requests. Will be called with an argument of [/datum/http_response] as first argument
+	var/datum/callback/cb
+	/// Number of retries due to Status 429
+	var/retry_count = 0
+	/// When this request should be retried
+	var/retry_at = 0
+
+/*
+###########################################################################
+THE METHODS IN THIS FILE ARE TO BE USED BY THE SUBSYSTEM AS A MANGEMENT HUB
+----------------------- DO NOT MANUALLY INVOKE THEM -----------------------
+###########################################################################
+*/
+
+/**
+  * Preparation handler
+  *
+  * Call this with relevant parameters to form the request you want to make
+  *
+  * Arguments:
+  * * method - HTTP Method to use, see code/__DEFINES/rust_g.dm for a full list
+  * * url - The URL to send the request to
+  * * body - The body of the request, if applicable
+  * * headers - Associative list of HTTP headers to send, if applicab;e
+  */
 /datum/http_request/proc/prepare(method, url, body = "", list/headers, output_file)
-	if (!length(headers))
-		headers = ""
+	if(!length(headers))
+		src.headers = ""
 	else
-		headers = json_encode(headers)
+		src.headers = json_encode(headers)
 
 	src.method = method
 	src.url = url
 	src.body = body
-	src.headers = headers
 	src.output_file = output_file
 
+/**
+  * Blocking executor
+  *
+  * Remains as a proof of concept to show it works, but should NEVER be used to do FFI halting the entire DD process up
+  * Async rqeuests are much preferred, but also require the subsystem to be firing for them to be answered
+  */
 /datum/http_request/proc/execute_blocking()
-	_raw_response = rustg_http_request_blocking(method, url, body, headers, build_options())
-
+	CRASH("Attempted to execute a blocking HTTP request")
+	// _raw_response = rustg_http_request_blocking(method, url, body, headers, build_options())
+
+/**
+  * Async execution starter
+  *
+  * Tells the request to start executing inside its own thread inside RUSTG
+  * Preferred over blocking, but also requires SShttp to be active
+  * As such, you cannot use this for events which may happen at roundstart (EG: IPIntel, BYOND account tracking, etc)
+  */
 /datum/http_request/proc/begin_async()
-	if (in_progress)
-		crash_with("Attempted to re-use a request object.")
+	if(in_progress)
+		CRASH("Attempted to re-use a request object.")
 
 	id = rustg_http_request_async(method, url, body, headers, build_options())
 
-	if (isnull(text2num(id)))
-		crash_with("Proc error: [id]")
+	if(isnull(text2num(id)))
 		_raw_response = "Proc error: [id]"
+		CRASH("Proc error: [id]")
 	else
 		in_progress = TRUE
 
+/**
+  * Options builder
+  *
+  * Builds options for if we want to download files with SShttp
+  */
```

Comment:
Should describe what the returns mean (eg. that it can return a list with format blah blah or null if the output fails/there's none/w.e

Reply:
This is only used as parameter for `rustg_http_request_async(method, url, body, headers, build_options())` if someone wants to modify that they need to look into the rust_g library to figure out what it expacts